config/initializers/puppet.rb

```diff
@@ -12,7 +12,7 @@
     :priority => 100
 end
 
-FactImporter.register_fact_importer :puppet, PuppetFactImporter, true
+Foreman::Plugin.importer_registry.register_importer :puppet, 'PuppetFactImporter', true
```

Comment:
You should be able to omit `importer_registry` and call `register_importer` on `Foreman::Plugin` directly, as the method will be delegated there.

Reply:
I tried doing that but it doesnt allow me to do so. I get an error: `/home/vagrant/foreman/config/initializers/puppet.rb:15:in <top (required)>: undefined method 'register_importer' for Foreman::Plugin:Class (NoMethodError)`

Reply:
Ah! I'm confusing this with when registering from a plugin. The delegation is then available as it's within an Plugin instance. That you can use in the Katello PR.

Reply:
So i guess `Foreman::Plugin.importer_registry.register_importer` will stay as is here?

Reply:
:+1: